hyperparameter_tuning/mxnet_gluon_cifar10_random_search/cifar10.py

```diff
@@ -0,0 +1,186 @@
+from __future__ import print_function
+
+import json
+import logging
+import os
+import time
+
+import mxnet as mx
+from mxnet import autograd as ag
+from mxnet import gluon
+from mxnet.gluon.model_zoo import vision as models
+
+
+# ------------------------------------------------------------ #
+# Training methods                                             #
+# ------------------------------------------------------------ #
+
+def train(current_host, hosts, num_cpus, num_gpus, channel_input_dirs, model_dir, hyperparameters, **kwargs):
+    # retrieve the hyperparameters we set in notebook (with some defaults)
+    batch_size = hyperparameters.get('batch_size', 128)
+    epochs = hyperparameters.get('epochs', 100)
+    learning_rate = hyperparameters.get('learning_rate', 0.1)
+    momentum = hyperparameters.get('momentum', 0.9)
+    log_interval = hyperparameters.get('log_interval', 1)
+    wd = hyperparameters.get('wd', 0.0001)
+
+    if len(hosts) == 1:
+        kvstore = 'device' if num_gpus > 0 else 'local'
+    else:
+        kvstore = 'dist_device_sync'
+
+    ctx = [mx.gpu(i) for i in range(num_gpus)] if num_gpus > 0 else [mx.cpu()]
+    net = models.get_model('resnet34_v2', ctx=ctx, pretrained=False, classes=10)
+    batch_size *= max(1, len(ctx))
+
+    # load training and validation data
+    # we use the gluon.data.vision.CIFAR10 class because of its built in pre-processing logic,
+    # but point it at the location where SageMaker placed the data files, so it doesn't download them again.
+
+    part_index = 0
+    for i, host in enumerate(hosts):
+        if host == current_host:
+            part_index = i
+            break
```

Comment:
What about 
```python
part_index = hosts.index(host)
```

Reply:
That does seem cleaner.  In this case, I copied `cifar10.py` right from the [existing example notebook](https://github.com/awslabs/amazon-sagemaker-examples/blob/master/sagemaker-python-sdk/mxnet_gluon_cifar10/cifar10.py#L41).  For consistency, I'll leave it as is.